sourcegraph/cody
# Edit: Maintain original insertion point when concurrent edits are being applied

## Description

This PR ensures we update the provided `range` when prompting a user for a edit instruction.

This avoids a scenario where:
1. User starts writing an edit instruction
2. Different edit applies somewhere, updating the same document, whilst the user is writing this instruction
3. User submits edit instruction, but the task has the wrong insertion point as the document has changed.

## Before
https://github.com/sourcegraph/cody/assets/9516420/71e1e26e-6e20-40b5-93b5-048460cbf06c


## After


https://github.com/sourcegraph/cody/assets/9516420/2ac6f3e4-4adb-4187-97ce-2d2035174df3


## Test plan

1. Execute an edit
2. Trigger a new edit before that one finishes applying, execute it
3. Check that BOTH edits are applied correctly.
